cdap-ui/app/cdap/components/DataPrepConnections/UploadFile/index.js

```
@@ -0,0 +1,202 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component, PropTypes } from 'react';
+import FileDnD from 'components/FileDnD';
+import NamespaceStore from 'services/NamespaceStore';
+import cookie from 'react-cookie';
+import isNil from 'lodash/isNil';
+import UploadFile from 'services/upload-file';
+import IconSVG from 'components/IconSVG';
+import T from 'i18n-react';
+
+require('./UploadFile.scss');
+
+const PREFIX = 'features.DataPrepConnections.UploadComponent';
+const FILE_SIZE_LIMIT = 10 * 1024 * 1024; // 10 MB
+
+export default class ConnectionsUpload extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      file: '',
+      recordDelimiter: '\\n',
+      error: null
+    };
+
+    this.fileHandler = this.fileHandler.bind(this);
+    this.recordDelimiterHandler = this.recordDelimiterHandler.bind(this);
+    this.upload = this.upload.bind(this);
+    this.clearError = this.clearError.bind(this);
+  }
+
+  fileHandler(e) {
+    let stateObj = {
+      file: e[0]
+    };
+
+    if (e[0].size > FILE_SIZE_LIMIT) {
+      stateObj.error = T.translate(`${PREFIX}.fileSizeError`);
+    }
+
+    this.setState(stateObj);
```

on reupload of correct file it would be good if we hide the error message.